src/services/staking_vaults.py

```diff
@@ -0,0 +1,659 @@
+        vaults: VaultsMap,
+        bs: ReferenceBlockStamp,
+        prev_tree_cid: str,
+        chain_config: ChainConfig,
+        vaults_fee_map: VaultFeeMap,
+    ) -> CID:
+        output = self.get_dumped_tree(
+            tree=tree,
+            vaults=vaults,
+            bs=bs,
+            prev_tree_cid=prev_tree_cid,
+            chain_config=chain_config,
+            vaults_fee_map=vaults_fee_map,
+        )
+
+        dumped_tree_str = json.dumps(output, default=self.tree_encoder)
+
+        return self.w3.ipfs.publish(dumped_tree_str.encode('ascii'), MERKLE_TREE_VAULTS_FILENAME)
+
+    @staticmethod
+    def get_dumped_tree(
+        tree: StandardMerkleTree,
+        vaults: VaultsMap,
+        bs: ReferenceBlockStamp,
+        prev_tree_cid: str,
+        chain_config: ChainConfig,
+        vaults_fee_map: VaultFeeMap,
+    ) -> dict[str, Any]:
+        def stringify_values(data) -> list[dict[str, str | int]]:
+            out = []
+            for item in data:
+                out.append({
+                    "value": (item["value"][0],) + tuple(str(x) for x in item["value"][1:]),
+                    "treeIndex": item["treeIndex"],
+                })
+            return out
+
+        values = stringify_values(tree.values)
+
+        extra_values = {}
+        for vault_adr, vault_info in vaults.items():
+            extra_values[vault_adr] = ExtraValue(
+                in_out_delta=str(vault_info.in_out_delta),
+                prev_fee=str(vaults_fee_map[vault_adr].prev_fee),
+                infra_fee=str(vaults_fee_map[vault_adr].infra_fee),
+                liquidity_fee=str(vaults_fee_map[vault_adr].liquidity_fee),
+                reservation_fee=str(vaults_fee_map[vault_adr].reservation_fee),
+            ).to_camel_dict()
+
+        output: dict[str, Any] = {
+            **dict(tree.dump()),
+            "refSlot": bs.ref_slot,
+            "blockHash": bs.block_hash,
+            "blockNumber": bs.block_number,
+            "timestamp": chain_config.genesis_time + bs.slot_number * chain_config.seconds_per_slot,
+            "extraValues": extra_values,
+            "prevTreeCID": prev_tree_cid,
+            "leafIndexToData": {k.value: v.value for k, v in MerkleValue.leaf_index_to_data().items()},
+        }
+        output.update(values=values)
+
+        return output
+
+    def get_ipfs_report(self, ipfs_report_cid: str, current_frame: FrameNumber) -> StakingVaultIpfsReport:
+        if ipfs_report_cid == "":
+            raise ValueError("Arg ipfs_report_cid could not be ''")
+        return self.get_vault_report(ipfs_report_cid, current_frame)
+
+    def get_vault_report(self, tree_cid: str, current_frame: FrameNumber) -> StakingVaultIpfsReport:
+        bb = self.w3.ipfs.fetch(CID(tree_cid), current_frame)
+        return StakingVaultIpfsReport.parse_merkle_tree_data(bb)
+
+    def get_latest_onchain_ipfs_report_data(self, block_identifier: BlockIdentifier) -> OnChainIpfsVaultReportData:
```

Comment:
lets remove this function. Since and replace with direct contract call. Also report is not from IPFS, so the name confuses a little bit